### ISSUE TITLE
Gatsby-plugin-sass no longer imports module styles as of 4.0.1

### ISSUE THREAD

**Issue by shennan** [gatsbyjs/gatsby#30037]
## Description

Gatsby sass css modules broken in V3. I have read the migration and implemented the new import syntax. When importing sass css modules á la:

```js
import * as styles from './SomeStyles.module.scss'

console.log(styles)

/*

Output:
{
  Symbol(Symbol.toStringTag): "Module"
  __esModule: true
  __proto__: Object
}
*/
```

There is effectively an empty Module in the output, with no style key/values even if the `SomeStyles.module.scss` has valid styles. I believe this as a result of the update to the Gatsby Sass Plugin that was pushed a few days ago, as returning to `gatsby-plugin-sass` version 4.0.0 (not 4.0.1) seems to fix issue.

### Steps to reproduce

Very easy to produce with a Gatsby starter and a simple `*.module.scss` file imported.

For ease, here's a temporary repo:

https://github.com/shennan/repro-gatsby-broken-sass-plugin

### Expected result

You should see key/values representing styles in the `styles` object after importing.

### Actual result

The `styles` object is empty, bar some inherited properties.

### Environment

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-6360U CPU @ 2.00GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 12.20.2 - ~/.nvm/versions/node/v12.20.2/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v12.20.2/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 69.0.1
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.0.1 => 3.0.2
    gatsby-plugin-sass: ^4.0.1 => 4.0.1
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```

**Comment by wardpeet** [comment]
@shennan We made an issue in gatsby-plugin-sass@3.0.1, if you revert to version 3.0.0, this goes away. I have a patch up.

**Comment by shennan** [comment]
Do you mean 4.0.1? 🤔

**Comment by pieh** [comment]
@shennan Ops, Yeah - regression introduced in `4.0.1` - but reverting to `4.0.0` helps - we will be reverting offender ( https://github.com/gatsbyjs/gatsby/pull/30038 ), but in meantime downgrading 1 patch version can get you unblocked

**Comment by vladar** [comment]
Published in `gatsby-plugin-sass@4.0.2`
